src/middlewared/middlewared/plugins/ports/ports.py

```diff
@@ -1,9 +1,21 @@
 import itertools
 
+from collections import defaultdict
+
 from middlewared.service import Service, ValidationErrors
 
+from .utils import WILDCARD_IPS
+
+
+SYSTEM_PORTS = [(wildcard, port) for wildcard in WILDCARD_IPS for port in [67, 123, 3702, 5353, 6000]]
+
 
-SYSTEM_PORTS = [67, 123, 3702, 5353, 6000]
+def check_ip_is_ipv6(ip: str) -> bool:
```

Comment:
Could we use the standard `ipaddress` module to determine the ip address? This could be simplified to 1 function instead of 2.

Also, this is not returning the "protocol". This is returning the IP version. Something like:

```
def get_ip_version(ip: str) -> int:
    return ipaddress.ip_interface(ip).version
```